File: desktop/src/lib/api/tvdb.ts
import { invoke } from '@tauri-apps/api/core'
import type { TvdbEpisode } from '$lib/types/tvdb'

export async function getTvdbEpisodes(
    anilistId: number,
    format: string | null
): Promise<TvdbEpisode[]> {
    return invoke<TvdbEpisode[]>('get_tvdb_episodes', {
        anilistId,
        format,
    })
}

export async function getTvdbBackground(
    anilistId: number,
    format: string | null
): Promise<string | null> {
    return invoke<string | null>('get_tvdb_background', {
        anilistId,
        format,
    })
}

File: app/src-tauri/src/commands/tvdb.rs
use crate::state::AppState;
use namizo_core::TvdbEpisode;
use tauri::State;

#[tauri::command]
pub async fn get_tvdb_episodes(
    anilist_id: u32,
    format: Option<String>,
    state: State<'_, AppState>,
) -> Result<Vec<TvdbEpisode>, String> {
    state
        .tvdb_service
        .get_episodes(anilist_id, format.as_deref())
        .await
}

#[tauri::command]
pub async fn get_tvdb_background(
    anilist_id: u32,
    format: Option<String>,
    state: State<'_, AppState>,
) -> Result<Option<String>, String> {
    state
        .tvdb_service
        .get_background(anilist_id, format.as_deref())
        .await
}

#[tauri::command]
pub async fn get_tvdb_clear_logo(
    anilist_id: u32,
    format: Option<String>,
    state: State<'_, AppState>,
) -> Result<Option<String>, String> {
    state
        .tvdb_service
        .get_clear_logo(anilist_id, format.as_deref())
        .await
}

#[tauri::command]
pub async fn clear_tvdb_cache(state: State<'_, AppState>) -> Result<(), String> {
    state.tvdb_service.clear_cache()
}